src/map/map.ts

```diff
@@ -128,38 +128,89 @@ function arrayMaxMin(values: number[]): {max: number, min: number} {
     return {max, min};
 }
 
-/** HTML element holding settings for a given axis (x, y, z, color) */
-class AxisSetting {
+interface AxisPresets {    
     /// Which property should we use for this axis
-    public property: HTMLSetting<'string'>;
+    property?: string;
     /// Which scale (linear/log) should we use
-    public scale: HTMLSetting<'string'>;
+    scale?: string;
     /// The minimal value for this axis
-    public min: HTMLSetting<'number'>;
+    min?: number;
     /// The maximal value for this axis
-    public max: HTMLSetting<'number'>;
+    max?: number;
+}
+
+const AxisDefaults = {
+     property : '',
+     scale : 'linear',
+     min : 0,
+     max: 0,
+}
 
-    constructor(properties: string[], initial: string) {
+/** HTML element holding settings for a given axis (x, y, z, color) */
+class AxisSettings {
+    public property!: HTMLSetting<'string'>;
+    public scale!: HTMLSetting<'string'>;
+    public min!: HTMLSetting<'number'>;
+    public max!: HTMLSetting<'number'>;
+
+    private _properties: string [];
+    
+    constructor(properties: string[], presets?: AxisPresets) {
+        this._properties = properties;
+        
+        // creates settings
         this.max = new HTMLSetting('number', 0);
         this.min = new HTMLSetting('number', 0);
-
-        this.property = new HTMLSetting('string', initial);
+        this.property = new HTMLSetting('string', '');
+        this.scale = new HTMLSetting('string', '');
+                
+        this.applyPresets(presets);
+        
         this.property.validate = (value) => {
-            if (properties.includes(value) || (initial === '' && value === '')) {
+            if (properties.includes(value) || value === '') {
```

Comment:
Here we loose the validation that prevent setting x/y to an empty string. We could keep both the inital (perhaps renamed to default) & preset arguments to be able to do this.

Reply:
I think that the fact that x and y axes are special should be dealt with in the MapSettings - in general we want to be able to have axes with no settings.